std/typecons.d

```diff
@@ -6896,3 +6896,222 @@ unittest
         ubyte, ubyte, T3, T3,
     );
 }
+
+/++
+ + Type constructor for head-const variables.
+ +
+ + Head-const variables cannot be directly mutated or rebound, but references
+ + reached through the variable are typed with their original mutability.
+ + It is equivalent to $(D final) variables in D1 and Java, as well as
+ + $(D readonly) variables in C#.
+ +
+ + When $(D T) is a $(D const) or $(D immutable) type, $(D HeadConst) aliases
+ + to $(D T).
+ +/
+struct HeadConst(T)
+    if (!is(T == const) && !is(T == immutable))
+{
+    private T HeadConst_value;
+    mixin Proxy!HeadConst_value;
+
+    /++
+     + Construction is forwarded to the underlying type.
+     +/
+    this(T other)
+    {
+        this.HeadConst_value = other;
+    }
+
+    /// Ditto
+    this(Args...)(auto ref Args args)
+        if (__traits(compiles, T(args)))
+    {
+        static assert((!is(T == struct) && !is(T == union)) || !isNested!T,
+            "Nested type " ~ fullyQualifiedName!T ~ " must be constructed " ~
```

Comment:
"*Nested type T must be* **marked as static** *or constructed explicitly at the call-site*", etc.

Since the error triggers even when the context is not actually being used for anything, just marking the type as `static` is going to be the best fix most of the time.